src/java/org/apache/cassandra/db/ColumnFamilyStore.java

```diff
@@ -2053,31 +2054,28 @@ public ColumnFamily getColumnFamily(QueryFilter filter)
 
             removeDroppedColumns(result);
 
+            DeletionInfo deletionInfo = result.deletionInfo();
```

Comment:
I was worried previously that the DeletionInfo would be recycled. However, upon further investigation, the deletion info should remain the same from when the query filter runs, and being referenced here.